Mathlib/Data/BitVec/Defs.lean

```diff
@@ -140,4 +140,10 @@ def toLEList (x : BitVec w) : List Bool :=
 def toBEList (x : BitVec w) : List Bool :=
   List.ofFn x.getMsb'
 
+instance : SMul ℕ (BitVec w) := ⟨fun x y => ofFin <| x • y.toFin⟩
+instance : SMul ℤ (BitVec w) := ⟨fun x y => ofFin <| x • y.toFin⟩
+instance : Pow (BitVec w) ℕ  := ⟨fun x n => ofFin <| x.toFin ^ n⟩
+instance : NatCast (BitVec w) := ⟨BitVec.ofNat w⟩
```

Comment:
I think this instance is already defined by Std? Maybe I'm mistaken

Reply:
These are all the instances that are defined in https://github.com/leanprover/std4/commit/fa17cba0459830906e99f0a81fb0edc106220380:

```bash
grep -R instance Std/Data/BitVec
Std/Data/BitVec/Basic.lean:instance : Inhabited (BitVec n) where default := .zero n
Std/Data/BitVec/Basic.lean:instance : OfNat (BitVec n) i where ofNat := .ofNat n i
Std/Data/BitVec/Basic.lean:instance : Repr (BitVec n) where reprPrec a _ := "0x" ++ (a.toHex : Format) ++ "#" ++ repr n
Std/Data/BitVec/Basic.lean:instance : ToString (BitVec n) where toString a := toString (repr a)
Std/Data/BitVec/Basic.lean:instance : Add (BitVec n) := ⟨BitVec.add⟩
Std/Data/BitVec/Basic.lean:instance : Sub (BitVec n) := ⟨BitVec.sub⟩
Std/Data/BitVec/Basic.lean:instance : Neg (BitVec n) := ⟨.neg⟩
Std/Data/BitVec/Basic.lean:instance : Mul (BitVec n) := ⟨.mul⟩
Std/Data/BitVec/Basic.lean:instance : Div (BitVec n) := ⟨.udiv⟩
Std/Data/BitVec/Basic.lean:instance : Mod (BitVec n) := ⟨.umod⟩
Std/Data/BitVec/Basic.lean:instance : LT (BitVec n) where lt x y := x.toFin < y.toFin
Std/Data/BitVec/Basic.lean:instance : LE (BitVec n) where le x y := x.toFin ≤ y.toFin
Std/Data/BitVec/Basic.lean:instance : AndOp (BitVec w) := ⟨.and⟩
Std/Data/BitVec/Basic.lean:instance : OrOp (BitVec w) := ⟨.or⟩
Std/Data/BitVec/Basic.lean:instance : Xor (BitVec w) := ⟨.xor⟩
Std/Data/BitVec/Basic.lean:instance : Complement (BitVec w) := ⟨.not⟩
Std/Data/BitVec/Basic.lean:instance : HShiftLeft (BitVec w) Nat (BitVec w) := ⟨.shiftLeft⟩
Std/Data/BitVec/Basic.lean:instance : HShiftRight (BitVec w) Nat (BitVec w) := ⟨.ushiftRight⟩
Std/Data/BitVec/Basic.lean:instance {n} : HShiftLeft  (BitVec m) (BitVec n) (BitVec m) := ⟨fun x y => x <<< y.toNat⟩
Std/Data/BitVec/Basic.lean:instance {n} : HShiftRight (BitVec m) (BitVec n) (BitVec m) := ⟨fun x y => x >>> y.toNat⟩
Std/Data/BitVec/Basic.lean:instance : HAppend (BitVec w) (BitVec v) (BitVec (w + v)) := ⟨.append⟩
Std/Data/BitVec/Basic.lean:instance : Subsingleton (BitVec 0) where
```

I do not see NatCast.

Reply:
I've confirmed with `#synth` that the instances at the least don't exist in the files imported at that point, which includes `Std.Data.BitVec`, so I'm pretty sure they don't exist in Std at all